### ISSUE TITLE
Add gamification features?

### ISSUE THREAD

**Issue by garciafdezpatricia** [Arquisoft/lomap_en2a#141]
I was looking at the optional requirements and one that looks very interesting is the one that includes gamification features. I was thinking of different options, what do you think?:

- Points and leaderboards. Award users points for adding locations to the map and display a leaderboard to show who has added the most locations (among their friends).
- Badges (SSI style). Create badges or achievements that users can earn for adding certain types of locations or reaching certain milestones, such as adding 5 locations, 10 locations, ...
- Progress Bars. Display a progress bar or percentage complete to show users how close they are to reaching a certain goal, such as adding a certain number of locations.
- Customizable Avatars. Allow users to customize their avatar based on their progress or accomplishments within the app, such as unlocking new items or outfits as they add more locations.

**Comment by UO283216** [comment]
I like the achivements part but I think the award and user avatar personalization is a little bit overkill . I do prefer to refactor existing functionallity to be much faster and/or user friendly than adding features that are not included in the optional requirements

**Comment by garciafdezpatricia** [comment]
The following proposals will be discarded:
- Realized that in order to have customized avatars we need a database to store the customization (not going to store it in the pod).
- Points and leaderboards collide with the control over the locations of our friends.
- Badges are not easy to do and there are other things that require my time and focus.

In addition, the following will be done:
- Different levels depending on the trophies that a user have.
- Trophies are achieved by adding locations, adding categories and reviews to the locations + additional trophies earned by completing challenges.

In summary, our gamification features are giong to be:
- Levels.
- Trophies.
- Fixed challenges.
